platform/broadcom/sonic-platform-modules-cel/tools/ispvme_12.2/hardware.c

```diff
@@ -0,0 +1,313 @@
+/*********************************************************************************
+* Lattice Semiconductor Corp. Copyright 2000-2008
+* 
+* This is the hardware.c of ispVME V12.1 for JTAG programmable devices.
+* All the functions requiring customization are organized into this file for 
+* the convinience of porting. 
+*********************************************************************************/
+/*********************************************************************************
+ * Revision History:
+ * 
+ * 09/11/07 NN Type cast mismatch variables
+ * 09/24/07 NN Added calibration function.
+ *             Calibration will help to determine the system clock frequency
+ *             and the count value for one micro-second delay of the target 
+ *             specific hardware.
+ *             Modified the ispVMDelay function
+ *             Removed Delay Percent support
+ *             Moved the sclock() function from ivm_core.c to hardware.c
```

Comment:
code license?

Reply:
Please refer to my new [commit.](https://github.com/Azure/sonic-buildimage/pull/2446/commits/4045360b628f7e582b967fc3c16c1f7d2be1c483)